lib/chef/resource/lwrp_base.rb

```diff
@@ -52,17 +52,37 @@ def build_from_file(cookbook_name, filename, run_context)
           resource_name = filename_to_qualified_string(cookbook_name, filename)
 
           # We load the class first to give it a chance to set its own name
-          resource_class = Class.new(self)
-          resource_class.resource_name resource_name.to_sym
-          resource_class.run_context = run_context
-          resource_class.class_from_file(filename)
-
-          # Make a useful string for the class (rather than <Class:312894723894>)
-          resource_class.instance_eval do
-            define_singleton_method(:to_s) do
-              "LWRP resource #{resource_name} from cookbook #{cookbook_name}"
+          deprecated_resource_class = Class.new(self).tap do |resource_class|
+            resource_class.resource_name(nil)
+            resource_class.run_context = run_context
+            resource_class.instance_eval do
+              define_method(:initialize) do |*args, &block|
+                Chef::Log::deprecation("Deprecated Thing") if chef_depreacted_access
```

Comment:
What if we did "if self.class == resource_class" instead? One less method to pollute the LWRP with ...